File: Python/hilos/TP6EJ2/Hilo.py
from threading import Thread, Lock

class Hilo(Thread):
    
    def __init__(self, contador):
        super().__init__()
        self.__lock = Lock()
        self.__contador = contador
    
    def run(self):
        for i in range(5000):
            with self.__lock:
                self.__contador.incrementar()

File: Java/TP6EJ2/src/Hilo.java

public class Hilo extends Thread{
	
	private Contador contador;
	
	public Hilo(Contador contador) {
		this.contador = contador;
	}
	
	@Override
	public void run() {
		for(int i = 0; i < 5000; i++) {
			this.contador.incrementar();
		}
	}

}
